### PR TITLE
Add warp reduce max/min test

### PR DESCRIPTION
**Pull request by abhilash1910**
Add tests for cub::warpreduce (cub::max/cub::min) for dpct.
cc @yihanwg

### CODE REVIEW

File: clang/test/dpct/cub/warplevel/warpreduce.cu [hunk 1]
```diff
@@ -68,6 +68,47 @@ __global__ void ReduceKernel(int* data) {
   data[threadid] = output;
 }
 
+
+//CHECK: void ReduceKernel_Max(int* data, const sycl::nd_item<3> &item_ct1) {
+//CHECK:  int threadid = item_ct1.get_local_id(2);
+//CHECK:  int input = data[threadid];
+//CHECK:  int output = 0;
+//CHECK:  output = sycl::reduce_over_group(item_ct1.get_sub_group(), input, sycl::maximum());
```
Comment (yihwang-nv) [inline clang/test/dpct/cub/warplevel/warpreduce.cu]:
```suggestion
//CHECK:  output = sycl::reduce_over_group(item_ct1.get_sub_group(), input, sycl::maximum<>());
```

File: clang/test/dpct/cub/warplevel/warpreduce.cu [hunk 2]
```diff
@@ -68,6 +68,47 @@ __global__ void ReduceKernel(int* data) {
...
+  output = WarpReduce(temp1).Reduce(input, cub::Max());
+  data[threadid] = output;
+}
+
+
+//CHECK: void ReduceKernel_Min(int* data, const sycl::nd_item<3> &item_ct1) {
+//CHECK:  int threadid = item_ct1.get_local_id(2);
+//CHECK:  int input = data[threadid];
+//CHECK:  int output = 0;
+//CHECK:  output = sycl::reduce_over_group(item_ct1.get_sub_group(), input, sycl::minimum());
```
Comment (yihwang-nv) [inline clang/test/dpct/cub/warplevel/warpreduce.cu]:
```suggestion
//CHECK:  output = sycl::reduce_over_group(item_ct1.get_sub_group(), input, sycl::minimum<>());
```

File: clang/test/dpct/cub/warplevel/warpreduce.cu [hunk 3]
```diff
@@ -141,6 +182,26 @@ int main() {
 //CHECK: q_ct1.parallel_for(
 //CHECK-NEXT:       sycl::nd_range<3>(GridSize * BlockSize, BlockSize),
 //CHECK-NEXT:       [=](sycl::nd_item<3> item_ct1) {{\[\[}}intel::reqd_sub_group_size(32){{\]\]}} {
+//CHECK-NEXT:         ReduceKernel_Max(dev_data, item_ct1)
```
Comment (yihwang-nv) [inline clang/test/dpct/cub/warplevel/warpreduce.cu]:
```suggestion
//CHECK-NEXT:         ReduceKernel_Max(dev_data, item_ct1);
```

File: clang/test/dpct/cub/warplevel/warpreduce.cu [hunk 4]
```diff
@@ -141,6 +182,26 @@ int main() {
...
+//CHECK-NEXT:       });
+  ReduceKernel_Max<<<GridSize, BlockSize>>>(dev_data);
+  cudaDeviceSynchronize();
+  verify_data(dev_data, TotalThread);  
+  
+  init_data(dev_data, TotalThread);
+//CHECK: q_ct1.parallel_for(
+//CHECK-NEXT:       sycl::nd_range<3>(GridSize * BlockSize, BlockSize),
+//CHECK-NEXT:       [=](sycl::nd_item<3> item_ct1) {{\[\[}}intel::reqd_sub_group_size(32){{\]\]}} {
+//CHECK-NEXT:         ReduceKernel_Min(dev_data, item_ct1)
```
Comment (yihwang-nv) [inline clang/test/dpct/cub/warplevel/warpreduce.cu]:
```suggestion
//CHECK-NEXT:         ReduceKernel_Min(dev_data, item_ct1);
```

**Review by yihwang-nv — CHANGES_REQUESTED** [review]
Need fix lit failures

File: clang/test/dpct/cub/warplevel/warpreduce.cu (hunk 3 above)
Comment (abhilash1910) [reply clang/test/dpct/cub/warplevel/warpreduce.cu]: Fixed lit

**Review by yihwang-nv — APPROVED** [review]
LGTM and let's wait CI green

**Review by ziranzha — APPROVED** [review]

File: clang/test/dpct/cub/warplevel/warpreduce.cu [hunk 5]
```diff
@@ -68,6 +68,47 @@ __global__ void ReduceKernel(int* data) {
   data[threadid] = output;
 }
 
+
+//CHECK: void ReduceKernel_Max(int* data, const sycl::nd_item<3> &item_ct1) {
+//CHECK:  int threadid = item_ct1.get_local_id(2);
+//CHECK:  int input = data[threadid];
+//CHECK:  int output = 0;
+//CHECK:  output = sycl::reduce_over_group(item_ct1.get_sub_group(), input, sycl::maximum<>());
```
Comment (ziranzha) [inline clang/test/dpct/cub/warplevel/warpreduce.cu]: Use CHECK-NEXT here maybe better to avoid ambiguous when check.

**Review by ziranzha — COMMENTED** [review]
Need sign-off for your commit.

File: clang/test/dpct/cub/warplevel/warpreduce.cu (hunk 5 above)
Comment (abhilash1910) [reply clang/test/dpct/cub/warplevel/warpreduce.cu]: Agree . added CHECK-NEXT for the other tests as well. Thanks

**Comment by abhilash1910** [comment]
Added signoff commit 
Signed-off-by: abhilash1910 <abhilash.majumder@intel.com>
@ziranzha  @yihanwg

File: clang/test/dpct/cub/warplevel/warpreduce.cu [hunk 6]
```diff
@@ -28,13 +28,16 @@ void print_data(int* data, int num) {
...
 }
 
-//CHECK: void SumKernel(int* data, const sycl::nd_item<3> &item_ct1) {
-//CHECK:  int threadid = item_ct1.get_local_id(2);
-//CHECK:  int input = data[threadid];
-//CHECK:  int output = 0;
-//CHECK:  output = sycl::reduce_over_group(item_ct1.get_sub_group(), input, sycl::plus<>());
-//CHECK:  data[threadid] = output;
-//CHECK:}
+//CHECK: void SumKernel(int* data,
```
Comment (yihwang-nv) [inline clang/test/dpct/cub/warplevel/warpreduce.cu]:
```suggestion
// CHECK: void SumKernel(int* data, const sycl::nd_item<3> &item_ct1) {
```

**Pull request merged**